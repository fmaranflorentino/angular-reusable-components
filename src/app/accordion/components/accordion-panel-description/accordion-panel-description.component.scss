.accordion_panel {
  &__description {
    font-size: 12px;
  }
}
