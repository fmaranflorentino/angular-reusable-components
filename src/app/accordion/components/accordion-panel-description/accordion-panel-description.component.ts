import { Component, OnInit, HostBinding, ViewEncapsulation } from '@angular/core';

@Component({
  selector: 'span[accordion-panel-description]',
  templateUrl: './accordion-panel-description.component.html',
  styleUrls: ['./accordion-panel-description.component.scss'],
  encapsulation: ViewEncapsulation.None,
})
export class AccordionPanelDescriptionComponent implements OnInit {
  @HostBinding('class.accordion_panel__header__description')
  accordionPanelDescription = true;

  constructor() { }

  ngOnInit(): void {
  }

}
