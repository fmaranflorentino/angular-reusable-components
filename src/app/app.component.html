<section>
  <button customButton color="primary">
    primary
  </button>

  <button customButton color="secondary">
    secondary
  </button>

  <button customButton outline>
    outline
  </button>

  <button customButton outline>
    outline icon left
    <img src="/assets/send.svg" alt="" icon />
  </button>

  <button customButton color="primary" iconPosition="right">
    icon right
    <img src="/assets/send.svg" alt="" icon />
  </button>

  <button customButton disabled>
    disabled
  </button>
</section>
<hr />
<accordion>
  <accordion-panel>
    <h3 panel-header-title>Angular - Developer Experience</h3>
    <span accordion-panel-description
      >A meetup event for defining the future of developer experience.</span
    >
    <span class="material-icons" panel-header-icon>
      account_circle
    </span>

    <accordion-panel-content>content man</accordion-panel-content>
  </accordion-panel>

  <accordion-panel>
    <h3 panel-header-title>Angular - Developer Experience</h3>
    <span accordion-panel-description
      >A meetup event for defining the future of developer experience.</span
    >
    <span class="material-icons" panel-header-icon>
      account_circle
    </span>

    <accordion-panel-content>content man</accordion-panel-content>
  </accordion-panel>
</accordion>
