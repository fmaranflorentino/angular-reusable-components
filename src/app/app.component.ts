import { Component } from '@angular/core';
// import { testeServ } from './accordion/components/bottom-sheet/teste';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent {
  // constructor(private teste: testeServ) {
  //   this.teste.open();
  // }
}
