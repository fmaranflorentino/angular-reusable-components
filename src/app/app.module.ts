import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { NgModule } from '@angular/core';

import { AppComponent } from './app.component';
import { ButtonComponent } from './button/button.component';
import { AccordionComponent } from './accordion/accordion.component';
import { AccordionPanelComponent } from './accordion/components/accordion-panel/accordion-panel.component';
import { AccordionPanelHeaderComponent } from './accordion/components/accordion-panel-header/accordion-panel-header.component';
import { AccordionPanelDescriptionComponent } from './accordion/components/accordion-panel-description/accordion-panel-description.component';
import { CommonModule } from '@angular/common';
import { AccordionPanelContentComponent } from './accordion/components/accordion-panel-content/accordion-panel-content.component';
import { BottomSheatComponent } from './accordion/components/bottom-sheat/bottom-sheat.component';
import { BottomSheetComponent } from './accordion/components/bottom-sheet/bottom-sheet.component';
// import { testeServ } from './accordion/components/bottom-sheet/teste';
// import { OVERLAY_PROVIDERS } from '@angular/cdk/overlay';
import { PanelHeaderTitleComponent } from './accordion/components/accordion-panel-header/components/panel-header-title/panel-header-title.component';

@NgModule({
  declarations: [
    AppComponent,
    ButtonComponent,
    AccordionComponent,
    AccordionPanelComponent,
    AccordionPanelHeaderComponent,
    AccordionPanelDescriptionComponent,
    AccordionPanelContentComponent,
    BottomSheatComponent,
    BottomSheetComponent,
    PanelHeaderTitleComponent,
  ],
  imports: [BrowserModule, BrowserAnimationsModule, CommonModule],
  providers: [
    // testeServ,
    // OVERLAY_PROVIDERS,
  ],
  bootstrap: [AppComponent],
  exports: [
    ButtonComponent,
    AccordionComponent,
    AccordionPanelComponent,
    AccordionPanelHeaderComponent,
    AccordionPanelDescriptionComponent,
    AccordionPanelContentComponent,
    BottomSheatComponent,
    BottomSheetComponent,
    PanelHeaderTitleComponent,
  ],
  entryComponents: [BottomSheetComponent],
})
export class AppModule {}
