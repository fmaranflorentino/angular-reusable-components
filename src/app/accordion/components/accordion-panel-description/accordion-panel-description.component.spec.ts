import { SpectatorHost, createHostFactory } from '@ngneat/spectator/jest';
import { axe } from 'jest-axe';

import { AppModule } from '../../../app.module';
import { APP_BASE_HREF } from '@angular/common';
import { AccordionPanelDescriptionComponent } from './accordion-panel-description.component';

const cases = {
  simple: `<span accordion-panel-description>description text</span>`,
};

describe('AccordionPanelDescriptionComponent', () => {
  let subject: SpectatorHost<AccordionPanelDescriptionComponent>;
  const createHost = createHostFactory({
    component: AccordionPanelDescriptionComponent,
    declareComponent: false,
    imports: [AppModule],
    providers: [{ provide: APP_BASE_HREF, useValue: '/' }],
  });

  it('should create instance', () => {
    subject = createHost(cases.simple);
    expect(subject.component).toBeTruthy();
  });

  it('should create and match raw component snapshot', () => {
    subject = createHost(cases.simple);
    expect(subject.hostFixture).toMatchSnapshot();
  });

  it('should have css class accordion_panel__header__description', () => {
    subject = createHost(cases.simple);
    expect(subject.hostFixture).toMatchSnapshot();
  });
});
