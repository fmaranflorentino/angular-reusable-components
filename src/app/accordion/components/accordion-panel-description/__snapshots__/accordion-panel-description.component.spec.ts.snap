// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionPanelDescriptionComponent should create and match raw component snapshot 1`] = `
<ng-component>
  <span
    accordion-panel-description=""
    class="accordion_panel__header__description"
  >
    description text
  </span>
</ng-component>
`;

exports[`AccordionPanelDescriptionComponent should have css class accordion_panel__header__description 1`] = `
<ng-component>
  <span
    accordion-panel-description=""
    class="accordion_panel__header__description"
  >
    description text
  </span>
</ng-component>
`;
